Automattic/woocommerce-payments
# Add fix for special case zero decimal currencies

Fixes #6946 

#### Changes proposed in this Pull Request
- Modified `prepare_amount` to handle special case zero-decimal currencies 

UGX and ISK are the special cases of zero decimal currencies as described [here](https://stripe.com/docs/currencies#special-cases) by Stripe. These currencies, though technically zero-decimal, have to be sent with two decimals to Stripe.

#### Testing instructions
- Setup a store with base currency UGX. Ensure that the products are setup with correct UGX prices (e.g. 18 UGX is not a valid price. The UGX to USD conversion rate is 4000)
- Use multi-currency module to checkout with UGX/ISK.
- Test checkout for different types of currencies - zero-decimal, special case (UGX, ISK) and two decimal currencies.
- Test Transaction, Order Edit and other screens if the values shown are correct.

-------------------


**Post merge**
